frontend/demo/component/crud/crud-open-editor.ts

```diff
@@ -0,0 +1,50 @@
+import '../../init'; // hidden-full-source-line
+import '@vaadin/flow-frontend/gridConnector.js'; // hidden-full-source-line
+
+import { html, LitElement, customElement, internalProperty, query } from 'lit-element';
+import '@vaadin/vaadin-crud/vaadin-crud';
+import { getPeople } from '../../domain/DataService';
+import Person from '../../../generated/com/vaadin/demo/domain/Person';
+import { applyTheme } from 'generated/theme';
+import { GridElement, GridEventContext } from '@vaadin/vaadin-grid';
+
+@customElement('crud-open-editor')
+export class Example extends LitElement {
+  constructor() {
+    super();
+    // Apply custom theme (only supported if your app uses one)
+    applyTheme(this.shadowRoot);
+  }
+
+  @internalProperty()
+  private items: Person[] = [];
+
+  @query('vaadin-grid')
+  private grid?: GridElement;
+
+  async firstUpdated() {
+    this.items = await getPeople();
+  }
+
+  render() {
+    return html`
+      <!-- tag::snippet[] -->
+      <vaadin-crud include="firstName, lastName, email, profession" .items=${this.items}>
+        <vaadin-grid slot="grid" @dblclick="${this.onDblClick}">
+          <vaadin-grid-column path="firstName" header="First name"></vaadin-grid-column>
+          <vaadin-grid-column path="lastName" header="Last name"></vaadin-grid-column>
+          <vaadin-grid-column path="email" header="Email"></vaadin-grid-column>
+          <vaadin-grid-column path="profession" header="Profession"></vaadin-grid-column>
+        </vaadin-grid>
+      </vaadin-crud>
+      <!-- end::snippet[] -->
+    `;
+  }
+
+  onDblClick(e: MouseEvent) {
+    const item = (this.grid?.getEventContext(e) as GridEventContext).item;
+    if (item) {
+      this.grid?.dispatchEvent(new CustomEvent('edit', { detail: { item: item } }));
+    }
+  }
```

Comment:
This is too hacky. Crud has `editedItem` property, that you should use instead
```suggestion
  onDblClick(e: MouseEvent) {
    this.editedItem = ((e.target as GridElement).getEventContext(e) as GridEventContext).item;
  }
```

Reply:
There is a problem with editing the same item. Unfortunately `editor-opened-changed` event cannot be used in this case either as it's fired before `editedItem` is changed on closing the editor.

Have created corresponding issue and attached a PR right away: https://github.com/vaadin/vaadin-crud/issues/241.
This is an enhancement, other option is to make sure `editedItem` is changed before emitting `editor-opened-changed`. Not sure which one is preferable.

Reply:
It's a shame that `editedItem` doesn't notify. Making it notify could almost be considered a bugfix as it's more like a missing event.

The following logic should work until we have the proper notify event available:
```js
@editor-opened-changed=${(e: CustomEvent) => {
  if (e.detail.value) {
    // The editor was opened
    this.editedItem = (e.currentTarget as CrudElement<Person>).editedItem;
  } else {
    this.editedItem = null;
  }
}}
```

or a shorter version:
```js
const editedItem = (e.currentTarget as CrudElement<Person>).editedItem;
this.editedItem = e.detail.value ? editedItem : null;
```

Reply:
The `editedItem` notifies now, example updated.

Reply:
Seems to have an error

<img width="906" alt="Screenshot 2021-04-12 at 16 10 18" src="https://user-images.githubusercontent.com/1222264/114399794-fceb7600-9ba9-11eb-8c75-77d596593b5c.png">

Reply:
Fixed the error.